metadata/metadata.go

```diff
@@ -1663,6 +1768,43 @@ func (serv *MetadataServer) genericList(t ResourceType, send sendFn) error {
 	return nil
 }
 
+func (serv *MetadataServer) getEquivalent(resourceVariant *pb.ResourceVariant) (*pb.ResourceVariant, error) {
+	var currentResource ResourceVariant
+	var resourceType ResourceType
+
+	switch resourceVariant.Resource.(type) {
+	case *pb.ResourceVariant_SourceVariant:
+		currentResource = &sourceVariantResource{resourceVariant.GetSourceVariant()}
+		resourceType = SOURCE_VARIANT
+	case *pb.ResourceVariant_FeatureVariant:
+		currentResource = &featureVariantResource{resourceVariant.GetFeatureVariant()}
+		resourceType = FEATURE_VARIANT
+	case *pb.ResourceVariant_LabelVariant:
+		currentResource = &labelVariantResource{resourceVariant.GetLabelVariant()}
+		resourceType = LABEL_VARIANT
+	case *pb.ResourceVariant_TrainingSetVariant:
+		currentResource = &trainingSetVariantResource{resourceVariant.GetTrainingSetVariant()}
+		resourceType = TRAINING_SET_VARIANT
+	}
+
+	resources, err := serv.lookup.ListForType(resourceType)
```

Comment:
@crfeliz or @anthonylasso is there a way to have auth around this call (or maybe some other call that I can use) so that we only get all resources given the users auth creds

Reply:
yeah in `main.go` where this is invoked, we can use `authorizer.GetUserResources()` to list them first.

Reply:
It looks like that function should probably have a filter by resource type

Reply:
yup